docs/architecture/overview.md

```
@@ -0,0 +1,64 @@
+│   │   ├── ConnectedWorkerPool.tsx # Worker pool with context
+│   │   └── ...                   # Other UI components
+│   ├── App.tsx                   # Composition root
+│   └── main.tsx                  # Application entry point
+├── docs/                         # Documentation
+└── package.json                  # Project definition
+```
+
+## Layer Responsibilities
+
+| Layer | Contains | Depends On |
+|-------|----------|------------|
```

markdownlint-cli2 (0.20.0)

[warning] Table column style
Table pipe is missing space to the right for style "compact"

(MD060, table-column-style)